zebra-network/src/meta_addr/tests/vectors.rs

```diff
@@ -34,3 +45,129 @@ fn sanitize_extremes() {
+}
+
+/// Test gossiped peer that is recently reachable.
+#[test]
+fn gossiped_peer_reportedly_to_be_seen_recently_is_recently_reachable() {
+    zebra_test::init();
+
+    let address = SocketAddr::from(([192, 168, 180, 9], 10_000));
+
+    // Report last seen within the reachable interval.
+    let offset = REACHABLE_PEER_DURATION
+        .checked_sub(TEST_TIME_ERROR_MARGIN)
+        .expect("Test margin is too large");
+    let last_seen = DateTime32::now()
+        .checked_sub(offset)
+        .expect("Offset is too large");
+
+    let peer = MetaAddr::new_gossiped_meta_addr(address, PeerServices::NODE_NETWORK, last_seen);
+
+    assert!(peer.was_recently_reachable());
+}
+
+/// Test gossiped peer that was reportedly last seen in the future.
+#[test]
+fn gossiped_peer_reportedly_seen_in_the_future_is_recently_reachable() {
+    zebra_test::init();
+
+    let address = SocketAddr::from(([192, 168, 180, 9], 10_000));
+
+    // Report last seen in the future
+    let last_seen = DateTime32::now()
+        .checked_add(REACHABLE_PEER_DURATION)
+        .expect("Reachable peer duration is too large");
+
+    let peer = MetaAddr::new_gossiped_meta_addr(address, PeerServices::NODE_NETWORK, last_seen);
+
+    assert!(peer.was_recently_reachable());
+}
+
+/// Test gossiped peer that is not recently reachable.
+#[test]
+fn gossiped_peer_reportedly_seen_long_ago_is_not_recently_reachable() {
+    zebra_test::init();
+
+    let address = SocketAddr::from(([192, 168, 180, 9], 10_000));
+
+    // Report last seen just outside the reachable interval.
+    let offset = REACHABLE_PEER_DURATION
+        .checked_add(TEST_TIME_ERROR_MARGIN)
+        .expect("Test margin is too large");
+    let last_seen = DateTime32::now()
+        .checked_sub(offset)
+        .expect("Offset is too large");
+
+    let peer = MetaAddr::new_gossiped_meta_addr(address, PeerServices::NODE_NETWORK, last_seen);
+
+    assert!(!peer.was_recently_reachable());
+}
+
+/// Test that recently responded peer is recently reachable.
+#[test]
+fn recently_responded_peer_is_recently_reachable() {
+    zebra_test::init();
+
+    let address = SocketAddr::from(([192, 168, 180, 9], 10_000));
+    let peer_seed = MetaAddr::new_alternate(&address, &PeerServices::NODE_NETWORK)
+        .into_new_meta_addr()
+        .expect("MetaAddrChange can't create a new MetaAddr");
+
+    // Create a peer that has responded
+    let mut peer = MetaAddr::new_responded(&address, &PeerServices::NODE_NETWORK)
+        .apply_to_meta_addr(peer_seed)
+        .expect("Failed to create MetaAddr for responded peer");
+
+    // Tweak the peer's last response time to be within the limits of the reachable duration
+    let offset = REACHABLE_PEER_DURATION
+        .checked_sub(TEST_TIME_ERROR_MARGIN)
+        .expect("Test margin is too large");
+    let last_response = DateTime32::now()
+        .checked_sub(offset)
+        .expect("Offset is too large");
+
+    peer.set_last_response(last_response);
```

Comment:
Test Coverage: Can you also do the responded test without the tweak?

Reply:
Done in [20cbde6](https://github.com/jvff/zebra/commit/20cbde6bd960dfd2ee9465a85a7c539ff07c1721). I'm not sure if the test names are good, let me know if you have any ideas.

Reply:
I think those names are fine